flight/targets/naze32/link_STM32103CB_Naze32_sections.ld

```diff
@@ -0,0 +1,173 @@
+/* This is the size of the stack for all FreeRTOS IRQs */
+_irq_stack_size = 0x240;
+/* This is the size of the stack for early init: life span is until scheduler starts */
+_init_stack_size = 0x200;
+/* This is the size of the pre-rtos heap */
+_init_heap_size = 53 * 256;
+
+/* Stub out these functions since we don't use them anyway */
+PROVIDE (   vPortSVCHandler = 0 ) ;
+PROVIDE (   xPortPendSVHandler = 0 ) ;
+PROVIDE (   xPortSysTickHandler = 0 ) ;
+
+/* Section Definitions */
+SECTIONS
+{
+    .text :
+    {
+        PROVIDE (pios_isr_vector_table_base = .);
+        KEEP(*(.isr_vector .isr_vector.*))
+        *(.text .text.* .gnu.linkonce.t.*)
+        *(.glue_7t) *(.glue_7)
+        *(.rodata .rodata* .gnu.linkonce.r.*)
+    } > FLASH
+
+    /* module sections */
+    .initcallmodule.init :
+    {
+        . = ALIGN(4);
+        __module_initcall_start = .;
+        KEEP(*(.initcallmodule.init))
+        . = ALIGN(4);
+        __module_initcall_end   = .;
+    } >FLASH
+
+    /* 
+     * _exti section
+     */
+    _exti :
+    {
+        . = ALIGN(4);
+        __start__exti = .;
+        KEEP(*(_exti))
+        . = ALIGN(4);
+        __stop__exti   = .;
+    } >FLASH
+
+    .ARM.extab :
+    {
+        *(.ARM.extab* .gnu.linkonce.armextab.*)
+    } > FLASH
+
+    .ARM.exidx :
+    {
+        *(.ARM.exidx* .gnu.linkonce.armexidx.*)
+    } > FLASH
+
+    .boardinfo :
+    {
+        PROVIDE(pios_board_info_blob = .);
+        . = ALIGN(4);
+        KEEP(*(.boardinfo))
+        . = ALIGN(4);
+    } > FLASH	
+
+    /*
+     * Markers for the end of the 'text' section and the in-flash start of 
+     * non-constant data 
+     */
+    . = ALIGN(4);
+    _etext = .;
+    _sidata = .;
+
+    /*
+     */
+    .fw_version_blob :
+    {
+        KEEP(*(.fw_version_blob))
```

Comment:
i have code in my ESC branch that is a Qt plugin that implements the stm32flash stuff. With the ESC i had code in the boot loader that would allow it to reset itself into DFU mode.